index.js

```diff
@@ -30,31 +32,52 @@ function consulClient(host) {
     var requestOptions = prepareRequestOptions(settings);
     var healthUrl = composeConsulHealthUrl(
       host,
-      settings.serviceName,
-      settings.version
+      settings.serviceName
     );
+    var version = semver.validRange(settings.version);
 
     // Make a request to Consul host to retrieve service health info
     // selects a healthy service, if one exists
     // uses service data found to compose a url of where to reach the service
     // makes an HTTP request to the service.
     return got(healthUrl, { json: true })
-      .then(selectServiceInstance)
+      .then(function(response) {
+        return selectServiceInstance(response, version)
+      })
       .then(serviceUrlComposer(settings.endpoint))
       .then(makeRequest);
 
     /**
      * Selects a healthy service instance to use if one exists.
      *
      * @param {object} response
+     * @param {String} version
      * @returns {Service|*}
      */
-    function selectServiceInstance(response) {
+    function selectServiceInstance(response, version) {
       if (isEmpty(response.body)) {
         throw new Error('no service instances available');
       }
 
-      return sample(response.body).Service;
+      if (!version) {
+        throw new Error('invalid version supplied');
+      }
+
+      var matches = [];
+
+      response.body.forEach(function(hit) {
+        if (semver.satisfies(hit.Service.Tags[0] = hit.Service.Tags[0].replace(/-/g, '.'), version)) {
+          matches.push(hit.Service);
+        }
+      });
+
+      if(isEmpty(matches)) {
+        throw new Error('no services matching requested version were found');
```

Comment:
This is a little bit outside the scope of this ticket, but it drives me a bit crazy: could you go through and capitalize the first letter of the error messages here? (e.g. `No services matching the requested version were found`) They don't need to be perfect, but it would be nice if they felt a little more like ... sentences.